### DISCUSSION TITLE
Export code with modules

### DISCUSSION THREAD

**Discussion by dinolink22** [keplerproject/luarocks#1452] [Q&A]
I was finally able to install lua, luarocks and some rocks globally in the system, in their default paths.
I can use them via require so that works.
Now, really newbie question, how do i export my program but also the relative modules so it can be shared and used without the other person having luarocks or anything like that but in pure .lua?
I tried looking in the issues and discussions also on google but didn't really find an answer...
Thanks for the help!

**Comment by siffiejoe** [comment]
This is not something that LuaRocks does. You need something like a bundler. I would suggest [this one](https://github.com/siffiejoe/lua-amalg) (because I have written it), but you can find a list of links to other options at the top of the README.

**Reply by dinolink22** [reply]
Awesome Thanks! I'm gonna try your bundle today hopefully the guide will be easy to understand for me. If not i will ask for your help ^^

**Reply by dinolink22** [reply]
@siffiejoe I like your program, hopefully you can guide me on this little issue here and it's all good hopefully. (I'm using ubuntu, just in case it matters)
So I used this command:

`lua amalg.lua -o out.lua -s test.lua http.request`

and i can see my out.lua file with http.request file code and my test.lua code at the bottom.
It does not work on a machine without luarocks and the modules so after looking at the code I can see that http.request uses like roughly 15 other "requires" modules. I then read your docs and tried these commands: 

`lua -lamalg test.lua`
`lua amalg.lua -o out.lua -s test.lua -c`
`lua amalg.lua -o out1.lua -s test.lua -c -x`

The first to create the cache and then the second one it does captures more modules but it still does not work on another machine.
The file created with the 3rd command fails because it has C dlls or something i don't know.

How do i import a required modules with its sub-modules and if these sub-modules use some other sub-sub-modules import all those as well? Thanks!

**Reply by alerque** [reply]
1. Please read the docs. The readme covers your problem and explicitly states it doesn't scan for dependencies, you have to come up with a list yourself.
2. Please take this discussion to the related repository. This is all off topic here.
